src/main/java/ch/ivyteam/ivy/maven/deploy/DeployToEngineMojo.java

```diff
@@ -124,7 +125,7 @@ public class DeployToEngineMojo extends AbstractDeployMojo {
   @Parameter(property = "ivy.deploy.engine.url", defaultValue = HTTP_ENGINE_URL_DEFAULT)
   String deployEngineUrl;
 
-  @Component(role = org.sonatype.plexus.components.sec.dispatcher.SecDispatcher.class, hint = "default")
+  @Inject
   private SecDispatcher secDispatcher;

```

Comment:
@alexsuter fyi: maybe this could also be a solution to rewrite out plugin harness tests? ... there we always wrestle with setting up the Maven infra pieces, if Injection works also there, the setup should be easier.